### PR TITLE
pkp#28: provide backend header support [3.3]

### LINKED ISSUE #28: Provide support for editorial backend injection as well
ctgraham writes:

> We have the Custom Headers Plugin to allow arbitrary javascript/css/meta customization of the public interface, but no parallel for the backend editorial interface.  I have a use-case requesting the ability to load a javascript widget in the editorial interface which I think makes sense, and I think the general principle is broadly applicable.  I'm picturing Pitt ULS creating a maintaining an "Editorial Custom Headers Plugin" in the Plugin Gallery.  Any reasons this has not be done to-date, or any reason why this wouldn't be a good idea?

natewr suggests:

> I suspect a PR for the custom headers plugin might be welcome. Just add a second text field for headers that should be loaded on the editorial backend.

asmecher affirms:

> Yup, that would be a no-fuss solution.

### PR DESCRIPTION
**Pull request by ctgraham**
Resolves #28 for 3.3; C.f. [forward port to main](https://github.com/pkp/customHeader/pull/30).

### CODE REVIEW

File: CustomHeaderSettingsForm.inc.php [hunk 1]
```diff
@@ -73,8 +76,25 @@ function execute(...$functionArgs) {
...
+	 * Validate that the input is well-formed XML
+	 * We want to avoid breaking the whole HTML page with an unclosed HTML attribute quote or tag
+	 * @param $input string
+	 * @return boolean
+	 */
+	function validateWellFormed($input) {
+		$libxml_errors_setting = libxml_use_internal_errors();
+		libxml_use_internal_errors(true);
+		libxml_clear_errors();
+		$xml = simplexml_load_string($input);
```
Comment (asmecher) [inline CustomHeaderSettingsForm.inc.php]: Just a thought -- is this going to be overly restrictive? It's requiring well-formed XML, but the field should be checking for well-formed HTML. I think the DOM tools might provide an alternative:

https://www.php.net/manual/en/domdocument.loadhtml.php

The idea would be:

1. Use the same `libxml_use_internal_errors` approach and return value to determine whether a fatal parsing error occurred
2. If it's flexible enough to allow parsing of documents that we don't want to permit, maybe re-serialize the HTML before storing in the setting to resolve that?

Comment (ctgraham) [reply CustomHeaderSettingsForm.inc.php]: Fair enough.  It's unlikely that someone would want to legitimately throw an unclosed `li` in the `head`, but I could see someone wanting to do an old-style `meta` tag perhaps.

Comment (asmecher) [reply CustomHeaderSettingsForm.inc.php]: @ctgraham, unfortunately it looks like this won't work well. See:

https://github.com/pkp/customHeader/issues/33

I'll follow up on that issue.

Comment (ctgraham) [reply CustomHeaderSettingsForm.inc.php]: Perhaps and additional checkbox or a second confirmation "submit" which bypasses the validation? A "I solemnly swear I am up to no good" prompt?

Comment (asmecher) [reply CustomHeaderSettingsForm.inc.php]: @ctgraham, that sounds OK to me (and I'm not sure we'll find a better solution). I did find an OK-looking work-around that gets the validator to pass without breaking the JS, which I described [here](https://github.com/pkp/customHeader/issues/33#issuecomment-1868090661).

Comment (lpanebr) [reply CustomHeaderSettingsForm.inc.php]: "I solemnly swear I am up to no good" 🧙‍♂️ totally yes! After all who doesn't know that "with great power comes great responsibility"?

**Pull request merged**